standard/tier/tier_utils.lua

```diff
@@ -0,0 +1,177 @@
+---@param mode 'tiers'|'tiertypes'
+---@return boolean
+function Tier.isValid(input, mode)
+	return String.isEmpty(input) or Tier.raw(input, mode) ~= nil
+end
+
+--- Converts input to (storage) value for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string|integer|nil
+function Tier.toValue(input, mode)
+	return (Tier.raw(input, mode) or {}).value
+end
+
+--- Converts input to display name for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toName(input, mode)
+	return (Tier.raw(input, mode) or {}).name
+end
+
+--- Converts input to short name for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toShortName(input, mode)
+	return (Tier.raw(input, mode) or {}).short
+end
+
+--- Converts input to (portal) page for tier/tierType
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toLink(input, mode)
+	return (Tier.raw(input, mode) or {}).link
+end
+
+--- Converts input to a tier/tierType category
+---@param input string|integer|nil
+---@param mode 'tiers'|'tiertypes'
+---@return string?
+function Tier.toCategory(input, mode)
+	return (Tier.raw(input, mode) or {}).category
+end
+
+--- Builds the display for a single tier/tierType
+---@param args {
+---		input: string|integer|nil,
+---		mode: 'tiers'|'tiertypes',
+---		short: boolean?,
+---		link: string|boolean|nil,
+---		sort: boolean?
+---	}
+---@return string?
+function Tier.displaySingle(args)
+	args = args or {}
+
+	local tierData = Tier.raw(args.input, args.mode)
+
+	if not tierData then return '' end
+
+	local display = args.short and tierData.short or tierData.name
+
+	if not display then return '' end
+
+	local hiddenSort = args.sort and tostring(HiddenSort.run(tierData.sort)) or ''
+
+	if Logic.readBool(args.link) and tierData.link then
+		return hiddenSort .. Page.makeInternalLink({}, display, tierData.link)
+	elseif String.isNotEmpty(args.link) then
+		return hiddenSort .. Page.makeInternalLink({}, display, args.link)
+	end
+
+	return hiddenSort .. display
+end
+
+--- Builds the display for a a tier/tierType combination
+---@param args table
+---@return string?
+function Tier.display(args)
+	args = args or {}
+
+	args.tier = args.tier or args.liquipediatier
+	args.tiertype = args.tiertype or args.liquipediatiertype
+
+	if String.isEmpty(args.tier) and String.isEmpty(args.tiertype) then
+		return ''
+	elseif String.isEmpty(args.tier) or String.isEmpty(args.tiertype) then
+		args.shortIfBoth = false
+	end
+
+	local tierDisplay = Tier.displaySingle(Tier.parseDisplayArgs(args, 'tier'))
+
+	local tierTypeDisplay = Tier.displaySingle(Tier.parseDisplayArgs(args, 'tiertype'))
+	if String.isEmpty(tierTypeDisplay) then
+		return tierDisplay
+	end
+
+	return tierTypeDisplay .. NON_BREAKING_SPACE .. '(' .. tierDisplay .. ')'
+end
```

Comment:
I don't think input parsing belongs in a display component like this one. It should be up to the caller to parse the input and pass in the parsed props.

Here's how I would structure the display components:

```suggestion
---@param args {
---		link: boolean?,
---		short: boolean?,
---		sort: boolean?,
---		tier: string,
---	}
function Tier.Text(props)
	local data = assert(Tier.getData(props.tier))

	local name = props.short and data.short or data.name
	local text = props.link and data.link
		and Page.makeInternalLink({}, name, data.link)
		or name

	local hiddenSort = props.sort and tostring(HiddenSort.run('A' .. data.index)) or nil

	return (hiddenSort or '') .. text
end

---@param args {
---		link: boolean?,
---		short: boolean?,
---		sort: boolean?,
---		tierType: string,
---	}
function TierType.Text(props)
	local data = assert(TierType.getData(props.tierType))

	local name = props.short and data.short or data.name
	local text = props.link and data.link
		and Page.makeInternalLink({}, name, data.link)
		or name

	local hiddenSort = props.sort and tostring(HiddenSort.run('B' .. data.index)) or nil

	return (hiddenSort or '') .. text
end

---@param args {
---		link: boolean?,
---		short: boolean|'ifBoth'|nil,
---		sort: boolean?,
---		tier: string,
---		tierType: string?,
---	}
function Tier.PairText(props)
	if props.short == 'ifBoth' then
		props.short = props.tierType ~= nil
	end

	if props.tierType then
		local tierText = Tier.Text(Table.merge(props, {sort = false}))
		return TierType.Text(props) .. NON_BREAKING_SPACE .. '(' .. tierText .. ')'
	else
		return Tier.Text(props)
	end
end
```

Reply:
several problems with that
- you need to make the callers all customizable since some wikis need to pass e.g. game to this function because their links depend on the game (or some other stuff)
- the idea was to allow a lpdb table to be passable to this hence the fallback for the input and the parse to split the input readable by the called displaySingle, could split that in 2 sep. called functions though
- you loose e.g. the functionality to have custom provided links

Reply:
moved the parse stuff into a sep function

Reply:
> you need to make the callers all customizable since some wikis need to pass e.g. game to this function because their links depend on the game (or some other stuff)

There are many ways to accommodate this. One is to add `props.TierText` and `props.TierTypeText` to `Tier.Pair`, and plug those in in the wiki-specific `Tier.Pair`.

> you loose e.g. the functionality to have custom provided links

Which wiki would custom links be used in?

Reply:
> Which wiki would custom links be used in?
wikis that have different games on the same wiki and have different portal pages based on the game
e.g. AoE